### PR TITLE
Invert implementations to run/stop oscillator

### LINKED ISSUE #2: Running and halt methods are inversed
Hello and thank you for the library.

I am testing my DS1307 module using your library and I discovered that when I do `driver.set_running()`, the module doesn't actually work. If we look in the datasheet, it is indeed the case (https://www.analog.com/media/en/technical-documentation/data-sheets/DS1307.pdf):
> Bit 7 of Register 0 is the clock halt (CH) bit. When this bit is set to 1, the oscillator is disabled. When cleared to 0, the oscillator is enabled.

So, the solution would be just to swap the implementations.

### PR DESCRIPTION
**Pull request by PaulRaUnite**
Fixes #2.

### CODE REVIEW

**Review by eldruin — COMMENTED** [review]
Nice catch, thank you!
Could you add an entry to the changelog and also invert Bit 7 accordingly in the [run/halt tests](https://github.com/eldruin/ds1307-rs/blob/master/tests/run_halt.rs)?

I see now that some methods in a dependency have become deprecated and I have to update those too for the CI to pass.

**Comment by eldruin** [comment]
I fixed the problems that made the CI fail.
Could you rebase this to master?

**Comment by PaulRaUnite** [comment]
> Could you add an entry to the changelog and also invert Bit 7 accordingly in the [run/halt tests](https://github.com/eldruin/ds1307-rs/blob/master/tests/run_halt.rs)?
> I fixed the problems that made the CI fail. Could you rebase this to master?

Done. I haven't noticed that I forgot about the `running` method, but now it is fixed too.

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Updated internal use of `chrono`.
 - Updated `embedded-hal-mock` dev-dependency.
 - Updated MSRV to 1.62.0.
+- [breaking-change] Swap implementations of `set_running` and `halt`, inverted output of `running` as CH bit was interpreted incorrectly.
```
Comment (eldruin) [inline CHANGELOG.md]:
```suggestion

### Fixed
- Swapped implementations of `set_running` and `halt` and fixed output level of `running`.
```

File: tests/run_halt.rs [hunk 2]
```diff
@@ -2,39 +2,39 @@ use embedded_hal_mock::i2c::Transaction as I2cTrans;
 mod common;
 use crate::common::{destroy, new, Register, ADDR};
 
-get_test!(running, running, true, trans_read!(SECONDS, [0b1000_0000]));
-get_test!(not_running, running, false, trans_read!(SECONDS, [0]));
+get_test!(running, running, true, trans_read!(SECONDS, [0]));
+get_test!(not_running, running, false, trans_read!(SECONDS, [0b1000_0000]));
```
Comment (eldruin) [inline tests/run_halt.rs]:
```suggestion
get_test!(
    not_running,
    running,
    false,
    trans_read!(SECONDS, [0b1000_0000])
);
```
Fixed formatting

**Review by eldruin — APPROVED** [review]
Looks good to me, thank you!

**Comment by eldruin** [comment]
For the record, I have released this fix in version 0.5.0

**Pull request merged**